relation/src/gadgets/logic.rs

```diff
@@ -0,0 +1,424 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
+//! Logic related circuit implementations
+
+use ark_ff::PrimeField;
+
+use crate::{
+    errors::CircuitError,
+    gates::{CondSelectGate, LogicOrGate},
+    BoolVar, Circuit, PlonkCircuit, Variable,
+};
+
+impl<F: PrimeField> PlonkCircuit<F> {
+    /// Constrain that `a` is true or `b` is true.
+    /// Return error if variables are invalid.
+    pub fn logic_or_gate(&mut self, a: BoolVar, b: BoolVar) -> Result<(), CircuitError> {
+        self.check_var_bound(a.into())?;
+        self.check_var_bound(b.into())?;
+        let wire_vars = &[a.into(), b.into(), 0, 0, 0];
+        self.insert_gate(wire_vars, Box::new(LogicOrGate))?;
+        Ok(())
+    }
+
+    /// Obtain a bool variable representing whether two input variables are
+    /// equal. Return error if variables are invalid.
+    pub fn is_equal(&mut self, a: Variable, b: Variable) -> Result<BoolVar, CircuitError> {
```

Comment:
I remember in the very beginning we name it `is_equal()` but later either because of clippy or some other reason we change to `check_equal()`

@zhenfeizhang @chancharles92  can you remember why we make the change previously? 
(altho this is minor imo, doesn't hurt changing back)

Reply:
This is for constancy with naming of comparison gates, like `is_lt`. 
The same issue occurs between `*_gate` and `enforce_*`. They essentially represent the same type of constraints. IMO the later one is more straightforward. But it doesn't make sense to rename `logic_or_gate` to `enforce_logic_or`. Or we could leave `logic_or_gate` as is, and change other ones to `enforce_*`?

Reply:
consistency is good and indeed some of our gadget naming is not. 

I also prefer `is_equal()`, just can't recall why we change it in the first place, I'd say go ahead and use `is_equal()`.

I also prefer `enforce_*()` over `*_gate()`

Reply:
Okay let me work on this change. Using `is_*` and `enforce_*`. Except for some arithmetic operations like `add_gate`.